// gatsby-i18n/lib/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Head", {
  enumerable: true,
  get: function get() {
    return _Head.default;
  }
});
Object.defineProperty(exports, "Redirect", {
  enumerable: true,
  get: function get() {
    return _Redirect.default;
  }
});
Object.defineProperty(exports, "Language", {
  enumerable: true,
  get: function get() {
    return _Language.default;
  }
});
Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function get() {
    return _Link.default;
  }
});
Object.defineProperty(exports, "I18nProvider", {
  enumerable: true,
  get: function get() {
    return _I18nContext.I18nProvider;
  }
});
Object.defineProperty(exports, "I18nConsumer", {
  enumerable: true,
  get: function get() {
    return _I18nContext.I18nConsumer;
  }
});

var _Head = _interopRequireDefault(require("./Head"));

var _Redirect = _interopRequireDefault(require("./Redirect"));

var _Language = _interopRequireDefault(require("./Language"));

var _Link = _interopRequireDefault(require("./Link"));

var _I18nContext = require("./I18nContext");
